packages/kbn-config/src/config_service.ts

```diff
@@ -210,22 +210,17 @@ export class ConfigService {
     );
   }
 
-  private validateConfigAtPath(path: ConfigPath) {
-    return this.getDistinctConfig(path).pipe(map((config) => this.validateAtPath(path, config)));
-  }
-
-  private getDistinctConfig(path: ConfigPath) {
-    this.markAsHandled(path);
```

Comment:
We were flagging configurations as handled every time we called `atPath`. That probably made sense in the initial implementation, but since we now validate the whole config at startups, there are now unnecessary calls.

The only calls to `markAsHandled` remaining are the ones done during config validation, and when invoking `isEnabledAtPath` to mark a disabled plugin's path as handled.

Reply:
Can we move `validateAtPath` to perform validation only once on every `config$` emit? Then we get rid of validation overhead on every config read operation. WDYT? Can be done in a separate PR.a

Reply:
Validating the config within the whole config obs 

`this.config$ = combineLatest([this.rawConfigProvider.getConfig$(), this.deprecations]).pipe(...`

May be tricky because we would need to already prepare each child observable here, one per schema (we don't want `this.config$` to error.).

In a similar way, ideally, `atPathSync` should only validate the config once and memoize the error/validation result until next emission.

Also, as you said on slack, the timing between emissions and schema registration needs to be taken into account here. When registering a schema, we need to validate the associated part of the config.

`PluginInitializerContext.config` is not an API that should be used a lot, even during plugin initialization (theoretically the config should only be read once, more or less, per plugin), so this should be minor. 

I agree that this is still a performance improvement though, I will create an issue.